### PR TITLE
fix wrong explanation of missing native function

### PR DESCRIPTION
**Pull request by JerryKwan**
fix wrong explanation of missing native function when publishing

## Motivation

Fix the issue https://github.com/starcoinorg/move/issues/14

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan
Run the following command
```
cargo test -p move-cli --test cli_testsuite
```
and you should see something like this
```
running 27 tests
test run_all::explain_arithmetic_failure/args.txt         ... ok
test run_all::cyclic_multi_module_publish/args.txt        ... ok
test run_all::build_modules_and_scripts/args.txt          ... ok
test run_all::gas_metering/args.txt                       ... ok
test run_all::build_include_exclude_stdlib/args.txt       ... ok
test run_all::explain_resource_already_exists/args.txt    ... ok
test run_all::publish_then_run/args.txt                   ... ok
test run_all::examples_published_dev_mode/args.txt        ... ok
test run_all::events_emit_view/args.txt                   ... ok
test run_all::doctor_with_stdlib/args.txt                 ... ok
test run_all::multi_module_publish/args.txt               ... ok
test run_all::explain_user_tx_script_abort/args.txt       ... ok
test run_all::print_stack_trace/args.txt                  ... ok
test run_all::explain_missing_resource/args.txt           ... ok
test run_all::verify_native_functions_in_publish/args.txt ... ok
test run_all::republish_module/args.txt                   ... ok
test run_all::use_named_address/args.txt                  ... ok
test run_all::module_publish_view/args.txt                ... ok
test run_all::explain_user_module_abort/args.txt          ... ok
test run_all::explain_stdlib_abort/args.txt               ... ok
test run_all::print_values/args.txt                       ... ok
test run_all::package_basics/args.txt                     ... ok
test run_all::offer_smoke/args.txt                        ... ok
test run_all::read_write_set/args.txt                     ... ok
test run_all::republish/args.txt                          ... ok
test run_all::generate_struct_layout/args.txt             ... ok
test run_all::concretize_read_write_set/args.txt          ... ok

test result: ok. 27 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

```

### CODE REVIEW

**Comment by JerryKwan** [comment]
@jolestar Would you please help to review? thank you

**Comment by jolestar** [comment]
> @jolestar Would you please help to review? thank you

Commit to move-language first? then pickup

**Comment by JerryKwan** [comment]
@jolestar 
Original Move does not have this  issue. Seems like explain_publish_error was removed by some commit of starcoin
The problem which original move have is prompting wrong messages when publish modules with missing native functions in bundle mode, I have created an issue on original move already. https://github.com/move-language/move/issues/438, and will send PR once this issue is resolved

File: language/tools/move-cli/src/sandbox/commands/publish.rs [hunk 1]
```diff
@@ -120,6 +131,11 @@ pub fn publish(
                         if let Err(err) = res {
                             // TODO (mengxu): explain publish errors in multi-module publishing
                             println!("Invalid multi-module publishing: {}", err);
+                            // find CompiledUnitWithSource by module_id
+                            // if let Some(unit) = module_map.get(module_id.name().as_str()) {
+                            //     // explain publish error
+                            //     explain_publish_error(err, state, unit)?;
+                            // }
```
Comment (jolestar) [inline language/tools/move-cli/src/sandbox/commands/publish.rs:138]: why comment these lines?

Comment (JerryKwan) [reply language/tools/move-cli/src/sandbox/commands/publish.rs:138]: They are commented out on purpose. 
Will remove the comment and add new test case to solve the problem described https://github.com/move-language/move/issues/438

**Pull request merged**